flake.nix

```diff
@@ -37,50 +38,63 @@
+          in
+          lib.mapAttrs
+            (name: p: {
+              haddock = false; # Because, this is end-user software. No need for library docs.
+              libraryProfiling = false; # Avoid double-compilation.
+              justStaticExecutables = p.cabal.executables != [ ]; # Reduce closure size
+            })
+            localPackages;
+      };
+
+      perSystem = { pkgs, lib, config, system, ... }: {
         cachix-push.cacheName = "srid";
+        _module.args = import inputs.nixpkgs {
+          inherit system;
+          overlays = [
+            (self: super: {
+              stork-emanote = config.packages.stork;
+            })
+          ];
+        };
 
         # haskell-flake configuration
         haskellProjects.default = {
+          debug = true;
           imports = [
+            inputs.self.haskellFlakeProjectModules.localDefaults
             inputs.ema.haskellFlakeProjectModules.output
           ];
           devShell.tools = hp: {
-            inherit (config.packages)
-              stork;
+            inherit (pkgs)
+              stork-emanote;
             treefmt = config.treefmt.build.wrapper;
           } // config.treefmt.build.programs;
-          overrides = with pkgs.haskell.lib;
-            let
-              # Remove the given references from drv's executables.
-              # We shouldn't need this after https://github.com/haskell/cabal/pull/8534
-              removeReferencesTo = disallowedReferences: drv:
-                drv.overrideAttrs (old: rec {
-                  inherit disallowedReferences;
-                  # Ditch data dependencies that are not needed at runtime.
-                  # cf. https://github.com/NixOS/nixpkgs/pull/204675
-                  # cf. https://srid.ca/remove-references-to
-                  postInstall = (old.postInstall or "") + ''
-                    ${lib.concatStrings (map (e: "echo Removing reference to: ${e}\n") disallowedReferences)}
-                    ${lib.concatStrings (map (e: "remove-references-to -t ${e} $out/bin/*\n") disallowedReferences)}
-                  '';
-                });
-            in
-            self: super: {
-              commonmark-extensions = super.commonmark-extensions_0_2_3_2;
-              emanote =
-                lib.pipe super.emanote [
-                  (lib.flip addBuildDepends [ config.packages.stork ])
-                  dontHaddock
-                  disableLibraryProfiling
-                  justStaticExecutables
-                  (removeReferencesTo [
-                    self.pandoc
-                    self.pandoc-types
-                    self.warp
-                  ])
-                ];
+
+          packages = {
+            commonmark-extensions.root = "0.2.3.2";
+            # ema.root = lib.mkForce "0.8.2.0"; #  lib.mkForce (inputs.ema2 + /ema);
+          };
+          settings = {
+            emanote = { name, self, super, pkgs, ... }: {
+              check = false;
```

Comment:
Is it possible to move these args to the top-level? viz.:

```nix
settings = { name, self, super, pkgs, ... }: {
  emanote = { 
    check = false; ...
```

I don't believe so, because AFAIU a module cannot have dynamic list of top-level options.